app/controllers/admin/namespaces_controller.rb

```diff
@@ -1,6 +1,10 @@
 class Admin::NamespacesController < Admin::BaseController
+  respond_to :html, :js
+
   def index
     @special_namespaces = Namespace.where(global: true)
-    @namespaces = Namespace.not_portus.where(global: false).page(params[:page])
+    namespaces = Namespace.not_portus.where(global: false)
+    @namespaces = namespaces.search(params[:filter]).page(params[:page])
```

Comment:
I'm not very experienced with rails... But in Django, the similar code create only one query on db...